presto-thrift-connector-api/src/test/java/com/facebook/presto/connector/thrift/api/TestReadWrite.java

```diff
@@ -103,36 +108,66 @@
     }
 
     @Test(invocationCount = 20)
-    public void testReadWriteSingleValue()
+    public void testSingleRowPageReadWrite()
             throws Exception
     {
-        testReadWrite(new Random(singleValueSeedGenerator.incrementAndGet()), 1);
+        testPageReadWrite(new Random(singleRowPageSeedGenerator.incrementAndGet()), 1);
     }
 
     @Test(invocationCount = 20)
-    public void testReadWriteColumnData()
+    public void testSingleRowRecordSetReadWrite()
             throws Exception
     {
-        Random random = new Random(columnDataSeedGenerator.incrementAndGet());
-        int records = random.nextInt(10000) + 10000;
-        testReadWrite(random, records);
+        testRecordSetReadWrite(new Random(singleRowRecordSetSeedGenerator.incrementAndGet()), 1);
     }
 
-    private void testReadWrite(Random random, int records)
+    @Test(invocationCount = 20)
+    public void testMultiRowPageReadWrite()
             throws Exception
     {
+        Random random = new Random(multiRowPageSeedGenerator.incrementAndGet());
+        testPageReadWrite(random, random.nextInt(10000) + 10000);
+    }
+
+    @Test(invocationCount = 20)
+    public void testMultiRowRecordSetReadWrite()
+            throws Exception
+    {
+        Random random = new Random(multiRowRecordSetSeedGenerator.incrementAndGet());
+        testRecordSetReadWrite(random, random.nextInt(10000) + 10000);
+    }
+
+    private void testPageReadWrite(Random random, int records)
+    {
+        testReadWrite(random, records, blocks -> {
+            List<PrestoThriftBlock> columnBlocks = new ArrayList<>(columns.size());
+            for (int i = 0; i < columns.size(); i++) {
+                columnBlocks.add(fromBlock(blocks.get(i), columns.get(i).getType()));
+            }
+            return new PrestoThriftPageResult(columnBlocks, records, null);
+        });
+    }
+
+    private void testRecordSetReadWrite(Random random, int records)
+    {
+        testReadWrite(random, records, blocks -> {
+            List<Type> types = columns.stream().map(ColumnDefinition::getType).collect(toImmutableList());
+            PageRecordSet inputRecordSet = new PageRecordSet(types, new Page(blocks.toArray(new Block[blocks.size()])));
+            return fromRecordSet(inputRecordSet);
+        });
+    }
+
+    private void testReadWrite(Random random, int records,
```

Comment:
no need for a line break (or put the all args on multiple lines)

Reply:
Done